pkg/operator/controller/controller_dns_node_resolver_daemonset.go

```diff
@@ -157,6 +159,7 @@ func desiredNodeResolverDaemonSet(clusterIP, clusterDomain, openshiftCLIImage st
 			UpdateStrategy: appsv1.DaemonSetUpdateStrategy{
 				Type: appsv1.RollingUpdateDaemonSetStrategyType,
 				RollingUpdate: &appsv1.RollingUpdateDaemonSet{
+					MaxSurge:       &maxSurge,
```

Comment:
Should `nodeResolverDaemonSetConfigChanged` check for changes to `Spec.UpdateStrategy`?

Reply:
It should, but that may be outside the scope for this fix, and we don't have any need for it right now (except maybe to override users' changes in case users try to modify it).  Should fixing it be part of this PR?

Reply:
Hm, good points.

One counter-point, however, is that the DNS daemonset is reconciled on updates to `Spec.UpdateStrategy`.
I think I am would be in favor of fixing this in this PR, or at the very least, adding a TODO comment. Thoughts?

Reply:
Fixed.